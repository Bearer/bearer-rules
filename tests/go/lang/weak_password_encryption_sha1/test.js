const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)

  test("bad", () => {
    const testCase = "bad.go"
    expect(invoke(testCase)).toMatchSnapshot()
  })

  test("ok", () => {
    const testCase = "ok.go"
    expect(invoke(testCase)).toMatchSnapshot()
  })
})
