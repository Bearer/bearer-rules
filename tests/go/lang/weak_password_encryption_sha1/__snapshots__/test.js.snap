// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`go_lang_weak_password_encryption_sha1 bad 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "326"
      ],
      "id": "go_lang_weak_password_encryption_sha1",
      "title": "Usage of weak hashing library on a password (SHA1)",
      "description": "## Description\\n\\nA weak hashing library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), MD5 and its predecessors are considered weak hash algorithms and therefore shouldn't be used.\\n\\n❌ Do not use encryption for passwords, wherever possible:\\n\\n\`\`\`go\\nsha1.Sum([]byte('password'))\\n\`\`\`\\n\\n✅ Instead, we recommend using sha256:\\n\\n\`\`\`go\\nsha256.Sum256([]byte('string'))\\n\`\`\`\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/go_lang_weak_password_encryption_sha1",
      "line_number": 14,
      "full_filename": "/tmp/bearer-scan/bad.go",
      "filename": ".",
      "data_type": {
        "category_uuid": "dd88aee5-9d40-4ad2-8983-0c791ddec47c",
        "name": "Passwords"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 14,
        "end": 14,
        "column": {
          "start": 25,
          "end": 38
        }
      },
      "sink": {
        "start": 14,
        "end": 14,
        "column": {
          "start": 19,
          "end": 39
        },
        "content": "h.Sum(user.Password)"
      },
      "parent_line_number": 14,
      "snippet": "h.Sum(user.Password)",
      "fingerprint": "d2bf48a480ce27df38fcb27b6bd76b47_0",
      "old_fingerprint": "1353d706a4f9e7cb2dc48170b5f97477_0",
      "code_extract": "\\tfmt.Printf(\\"%x\\", h.Sum(user.Password))"
    },
    {
      "cwe_ids": [
        "326"
      ],
      "id": "go_lang_weak_password_encryption_sha1",
      "title": "Usage of weak hashing library on a password (SHA1)",
      "description": "## Description\\n\\nA weak hashing library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), MD5 and its predecessors are considered weak hash algorithms and therefore shouldn't be used.\\n\\n❌ Do not use encryption for passwords, wherever possible:\\n\\n\`\`\`go\\nsha1.Sum([]byte('password'))\\n\`\`\`\\n\\n✅ Instead, we recommend using sha256:\\n\\n\`\`\`go\\nsha256.Sum256([]byte('string'))\\n\`\`\`\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/go_lang_weak_password_encryption_sha1",
      "line_number": 17,
      "full_filename": "/tmp/bearer-scan/bad.go",
      "filename": ".",
      "data_type": {
        "category_uuid": "dd88aee5-9d40-4ad2-8983-0c791ddec47c",
        "name": "Passwords"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 16,
        "end": 16,
        "column": {
          "start": 17,
          "end": 30
        }
      },
      "sink": {
        "start": 17,
        "end": 17,
        "column": {
          "start": 19,
          "end": 33
        },
        "content": "sha1.Sum(data)"
      },
      "parent_line_number": 17,
      "snippet": "sha1.Sum(data)",
      "fingerprint": "d2bf48a480ce27df38fcb27b6bd76b47_1",
      "old_fingerprint": "1353d706a4f9e7cb2dc48170b5f97477_1",
      "code_extract": "\\tfmt.Printf(\\"%x\\", sha1.Sum(data))"
    }
  ]
}"
`;

exports[`go_lang_weak_password_encryption_sha1 ok 1`] = `"{}"`;
